keanu-python/keanu/algorithm/sampling.py

```diff
@@ -107,16 +111,21 @@ def sample(net: BayesNet,
     network_samples: JavaObject = sampling_algorithm.get_sampler().getPosteriorSamples(
         probabilistic_model.unwrap(), vertices_unwrapped, draws).drop(drop).downSample(down_sample_interval)
 
-    vertex_samples = {
+    vertex_samples: sample_types = {
         Vertex._get_python_label(vertex_unwrapped): list(
             map(Tensor._to_ndarray,
-                network_samples.get(vertex_unwrapped).asList())) for vertex_unwrapped in vertices_unwrapped
+                network_samples.get(vertex_unwrapped).asList(),
+                itertools.repeat(True, len(network_samples.get(vertex_unwrapped).asList()))))
```

Comment:
Do you need to specify a length here at all?  itertools.repeat with no "n" param will just return the same thing over and over again.